### DISCUSSION TITLE
FTP server for Teensy 4.1

### DISCUSSION THREAD

**Discussion by WaveSimm** [micropython/micropython#15555] [m.iMX RT / Teensy 4.x]
HI guys,
I am so happy due to you guys.
I am looking for the ftp server lib for micropython. 
Is there for Teensy 4.1 and Micropython?
How can I find ethernet lib in detail?
Have a great weekend in adcance~

**Comment by andrewleech** [comment]
Hi, there's a few different FTP servers you could try listed in the FTP section on https://awesome-micropython.com/

**Comment by robert-hh** [comment]
Attached is a copy for the uftpd.py server modified to use a wired lan. The initial ftpserver would search of a WiFi network. Before starting the ftp server, lan has to be enabled. Tested with a Teensy 4.1. The documentation is at https://github.com/robert-hh/FTP-Server-for-ESP8266-ESP32-and-PYBD

[uftpd_lan.zip](https://github.com/user-attachments/files/16410246/uftpd_lan.zip)

**Reply by WaveSimm** [reply]
Great!! thanks a lot as always!!!
